src/scrollspy/scrollspy.ts

```diff
@@ -0,0 +1,42 @@
+import {Directive, ElementRef, Injectable, Input, OnDestroy, OnInit} from 'angular2/core';
+
+@Directive({selector: '[ngb-scroll-spy]', host: {'[class.active]': 'active'}, exportAs: 'ngbScrollSpy'})
+export class NgbScrollSpy implements OnDestroy, OnInit {
+  active: boolean;
+  @Input() ngbScrollSpy: string;
+
+  constructor(private scrollSpier: NgbScrollSpier) {}
+
+  ngOnDestroy(): void { this.scrollSpier.deregister(this); }
+
+  ngOnInit(): void { this.scrollSpier.register(this); }
+}
+
+@Directive({selector: '[ngb-scroll-target]', exportAs: 'ngbScrollTarget'})
+export class NgbScrollTarget implements OnInit {
+  constructor(private elementRef: ElementRef, private scrollSpier: NgbScrollSpier) {}
+
+  checkSpies(evt: MouseEvent) { this.scrollSpier.spies.forEach(spy => this.checkSpy(spy, evt)); }
+
+  checkSpy(spy: NgbScrollSpy, evt: MouseEvent) {
+    let element = this.elementRef.nativeElement;
+    let target = element.querySelector(spy.ngbScrollSpy);
+
+    if (!target) {
+      return;
+    }
+
+    spy.active = target.offsetTop <= element.scrollTop && target.offsetTop + target.offsetHeight >= element.scrollTop;
+  }
+
+  ngOnInit(): void { this.elementRef.nativeElement.addEventListener('scroll', (e) => this.checkSpies(e)); }
+}
+
+@Injectable()
+export class NgbScrollSpier {
+  public spies: Set<NgbScrollSpy> = <Set<NgbScrollSpy>>new Set();
```

Comment:
public is still a default, so I am not sure if it is really needed. Also, you don't need to do `Set<NgbScrollSpy>` twice, or at least, I hope so.